hcl2template/parser.go

```diff
@@ -92,13 +92,8 @@ func (p *Parser) parse(filename string, vars map[string]string) (*PackerConfig,
 
 	// Decode variable blocks so that they are available later on. Here locals
 	// can use input variables so we decode them firsthand.
-	{
-		for _, file := range files {
-			diags = append(diags, p.decodeInputVariables(file, cfg)...)
-		}
-		for _, file := range files {
-			diags = append(diags, p.decodeLocalVariables(file, cfg)...)
-		}
+	for _, file := range files {
+		diags = append(diags, cfg.decodeVariables(file)...)
```

Comment:
We want `locals` to be able to use `var.`, like so :

```hcl
locals {
    something = var.foo
}

variable "foo" {/*...*/}
```

But this will not work because at the time we pare `z`, `foo` doesn't exist yet, also files are sorted in lexicographical order which will have the same incidence.

Ah, I should have added unit tests for these cases I explained !

Reply:
That's true! I'm keeping both loops then like it was before.